GVRf/Framework/src/org/gearvrf/utility/Preconditions.java

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.utility;
+
+public final class Preconditions {
```

Comment:
1. `Preconditions` seems OK, if perhaps a bit 'academic'. I'm not sure what, if anything, would be any better: `Expectations`, `CheckParameters`, `ValidateParameters`?
2. Not clear why the class is `final` - I can certainly _imagine_ it having `protected` members at some point in the future.
3. Perhaps the class should be `abstract`, to keep anyone from ever `new` ing an instance ...